packages/core/src/rules/common/assertions/utils.ts

```diff
@@ -102,7 +103,9 @@ export function buildSubjectVisitor(
     for (const assert of asserts) {
       if (properties) {
         for (const property of properties) {
-          runAssertion(node[property], assert, location.child(property), report);
+          // we can have resolvable scalar so need to resolve value here.
+          const value = isRef(node[property]) ? resolve(node[property])?.node : node[property];
```

Comment:
I have an open question if we can detect that it is a `$ref`? For example, enforce `$ref` usage for path items (or anything else that you may want to enforce it with). So in other words, I would like to be able to assert `isRef: true`.

Reply:
Also, I would like to assert some things about the ref, like if it is a ref to a another file (as opposed to the components section of the same file).

Reply:
> I would like to be able to assert isRef: true.

We can do it I think. Needs some changes but it should be possible.

> I would like to assert some things about the ref, like if it is a ref to a another file (as opposed to the components section of the same file).

I think we need some syntax idea there then.

Reply:
I made a separate issue for it: https://github.com/Redocly/openapi-cli/issues/648